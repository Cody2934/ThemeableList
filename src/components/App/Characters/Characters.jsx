import React from 'react';
import PropType from 'prop-types';

const Characters = ({ image, name, species, status }) => (
  <>
    <image src={image} />
    <h1>{name}</h1>
    <p>{species}</p>
    <p>{status}</p>
  </>
);

Characters.propTypes = {
  image: PropType.string.isRequired,
  name: PropType.string.isRequired,
  species: PropType.string.isRequired,
  status: PropType.string.isRequired
};

export default Characters;

