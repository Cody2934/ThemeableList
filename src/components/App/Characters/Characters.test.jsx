import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

describe('Character component', () => {
  it('renders Character', () => {
    const wrapper = shallow(<Characters image="source of image" name="Cody" species="Human" status="Dead on the inside but most people do not know so it is okay" />);
    expect(wrapper).toMatchSnapshot();
  });
});
