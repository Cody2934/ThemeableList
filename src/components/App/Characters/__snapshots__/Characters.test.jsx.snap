// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<Fragment>
  <image
    src="source of image"
  />
  <h1>
    Cody
  </h1>
  <p>
    Human
  </p>
  <p>
    Dead on the inside but most people do not know so it is okay
  </p>
</Fragment>
`;
